src/client.rs

```diff
@@ -239,126 +236,85 @@ impl Tidal {
             .map_err(Into::into)
     }
 
-    //pub async fn delete(&self, url: &str, payload: &Value, etag: String) -> ClientResult<String> {
-        //self.api_call(Method::DELETE, url, Some(payload), Some(etag).await
-    //}
-
+    // The following functions are for backward compatibility only
+    //
     pub async fn search(&self, term: &str, limit: Option<u16>) -> ClientResult<TidalSearch> {
-        let url = "/search";
-        let limit = if let Some(limit) = limit { limit } else { 10 };
-        let mut params: HashMap<String, String> = HashMap::new();
-        params.insert("query".to_owned(), term.to_owned());
-        params.insert("limit".to_owned(), limit.to_string());
-        let result = self.get(&url, &mut params).await?;
-        Self::convert_result::<TidalSearch>(&result)
+        warn!("DEPRECATION WARNING!: This method will be deprecated in the next version. Please favor using .searches().find()");
```

Comment:
Maybe for these and the rest below, this is more appropriate? https://doc.rust-lang.org/edition-guide/rust-2018/the-compiler/an-attribute-for-deprecation.html

Reply:
@jpalvarezl I thought that flag would make the compiler stop, but I see it doesn't. will do 👍